### PR TITLE
Upgrade Apollo Server

### PR DESCRIPTION
**Pull request by aminlatifi**

### CODE REVIEW

File: src/types/ApolloContext.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Response } from 'express';
+import { Pool, ModuleThread } from 'threads';
+import { ProjectResolverWorker } from '../workers/projectsResolverWorker';
+import { JwtVerifiedUser } from '../services/authorizationServices';
+
+export interface ContextRequest {
+  user: JwtVerifiedUser;
+  auth: any;
```
Comment (CarlosQ96) [inline src/types/ApolloContext.ts]: Should we avoid using any? I think we can specify it like this.
```
export interface AuthContextRequest {
   token?: string;
   error?: any;
}
```
The error is kind of hard to define..... as it depends 100% on which graphql function is called with what message. Maybe something General

File: src/server/bootstrap.ts [hunk 2]
```diff
@@ -122,48 +126,30 @@ export async function bootstrap() {
...
-        try {
-          if (!req) {
-            return null;
-          }
+    const apolloServerPlugins = [
+      process.env.DISABLE_APOLLO_PLAYGROUND !== 'true'
+        ? ApolloServerPluginLandingPageGraphQLPlayground({
+            endpoint: '/graphql',
+          })
+        : ApolloServerPluginLandingPageDisabled(),
```
Comment (CarlosQ96) [inline src/server/bootstrap.ts:140]: This plugin sets the following page:
image

Comment (CarlosQ96) [reply src/server/bootstrap.ts:140]: Compared to simply disabling it like we had it before:
image

Comment (CarlosQ96) [reply src/server/bootstrap.ts:140]: Should we leave it like that or simple return some form of 404 not found like before? 

I don't think it's that important but I think a 404 is more appropriated. A disabled route shouldn't reveal any data.

**Review by CarlosQ96 — APPROVED** [review]
Hey Amin I tested:

Some queries manually in the graph playground everything seems (ok), all tests seem (ok), I saw you fixed a bug with the user context (ok),  adminbro seems unnafected (ok), thread pools initialized correctly when debugging them (ok) 

I think we can proceeed to test in staging.

I just left some low priority concerns in comments.

File: src/server/bootstrap.ts (hunk 2 above)
Comment (aminlatifi) [reply src/server/bootstrap.ts:140]: Good point, I don't know how to solve it in a short time. Let's do it in another issue.

**Comment by aminlatifi** [comment]
Thanks @CarlosQ96

**Review by mohammadranjbarz — APPROVED** [review]
Thanks @aminlatifi 
I just make the dependencies'versions fixed

**Pull request merged**